ofrak_core/src/ofrak/core/lz4.py

```diff
@@ -0,0 +1,332 @@
+
+    compression_level: Compression level to use (default: 0).
+        - Negative values: Fast acceleration (faster, less compression)
+        - 0-2: Minimum compression (default, all produce same output)
+        - 3: Minimum high-compression mode
+        - 4-16: Higher compression levels (16 is maximum)
+    """
+
+    compression_level: int = 0
+
+
+class Lz4Packer(Packer[Lz4PackerConfig]):
+    """
+    Pack data into a compressed LZ4 file using modern frame format.
+
+    Implementation repacks modern frame format preserving frame metadata.
+    Compression level can be specified via config (default: 0).
+    """
+
+    targets = (Lz4ModernData,)
+
+    async def pack(self, resource: Resource, config: Lz4PackerConfig = None):
+        """
+        Pack data into `Lz4ModernData` format.
+
+        :param resource: The LZ4 resource to pack
+        :param config: Optional configuration specifying compression level
+        """
+        if config is None:
+            config = Lz4PackerConfig()
+
+        lz4_child = await resource.get_only_child()
+        child_data = await lz4_child.get_data()
+
+        # Use stored compression settings from the view
+        lz4_view = await resource.view_as(Lz4ModernData)
+        content_checksum = lz4_view.content_checksum
+        block_checksum = lz4_view.block_checksum
+        block_size_id = lz4_view.block_size_id
+        block_linked = lz4_view.block_linked
+        store_size = lz4_view.content_size != 0
+
+        lz4_compressed = lz4.frame.compress(
+            child_data,
+            compression_level=config.compression_level,
+            content_checksum=content_checksum,
+            block_checksum=block_checksum,
+            block_size=block_size_id,
+            block_linked=block_linked,
+            store_size=store_size,
+        )
+
+        original_size = await resource.get_data_length()
+        resource.queue_patch(Range(0, original_size), lz4_compressed)
+
+
+class Lz4LegacyPacker(Packer[Lz4PackerConfig]):
+    """
+    Pack data into compressed LZ4 legacy format.
+
+    Legacy format supports compression levels via lz4.block.compress():
+    - Negative values: Fast mode with acceleration
+    - 0: Default compression
+    - 1-12: High compression mode
+    """
+
+    targets = (Lz4LegacyData,)
+
+    async def pack(self, resource: Resource, config: Lz4PackerConfig = None):
+        """
+        Pack data into `Lz4LegacyData` format.
+
+        :param resource: The LZ4 legacy resource to pack
+        :param config: Optional configuration specifying compression level
+        """
+        if config is None:
+            config = Lz4PackerConfig()
+
+        lz4_child = await resource.get_only_child()
+        child_data = await lz4_child.get_data()
+
+        # LZ4 legacy format uses 8 MB blocks (see https://github.com/lz4/lz4/blob/67a385a170d2dc331a25677e0d20d96eef0450c5/programs/lz4io.c#L86)
+        LEGACY_BLOCK_SIZE = 8 * (1 << 20)  # 8 MB
```

Comment:
```suggestion
        LEGACY_BLOCK_SIZE = 8 * 1024 * 1024  # 8 MB
```

Same comment as above about using KB squared. Once again, feel free to ignore.